src/masternodes/rpc_icxorderbook.cpp

```diff
@@ -0,0 +1,1497 @@
+        }
+        else if (order->orderType == CICXOrder::TYPE_EXTERNAL)
+        {
+            // if (submitexthtlc.amount * order->orderPrice / COIN != offer->amount )
+            //     throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("cannot make ext htlc with that amount, different amount necessary for offer (%s) - %s != %s!",
+            //                     offer->creationTx.GetHex(), ValueFromAmount(submitexthtlc.amount * order->orderPrice / COIN).getValStr(), ValueFromAmount(offer->amount).getValStr()));
+
+            if (!metaObj["receiveAddress"].isNull())
+                submitexthtlc.receiveAddress=DecodeScript(metaObj["receiveAddress"].getValStr());
+            else
+                throw JSONRPCError(RPC_INVALID_PARAMETER,"Invalid parameters, argument \"receiveAddress\" must be non-null");
+            if (!::IsMine(*pwallet, DecodeDestination(metaObj["receiveAddress"].getValStr())))
+                throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Address (%s) is not owned by the wallet", metaObj["receiveAddress"].getValStr()));
+
+            bool found = false;
+            pcustomcsview->ForEachICXSubmitEXTHTLCOpen([&found, &submitexthtlc](CICXOrderView::TxidPairKey const & key, uint8_t i) {
+                if (key.first != submitexthtlc.offerTx)
+                    return false;
+                found = true;
+                return false;
+            }, submitexthtlc.offerTx);
+            if (found)
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "ext htlc already submitted!");
+
+            found = false;
+            pcustomcsview->ForEachICXSubmitDFCHTLCOpen([&found, &submitexthtlc](CICXOrderView::TxidPairKey const & key, uint8_t) {
+                if (key.first == submitexthtlc.offerTx)
+                {
+                    found = true;
+                    return (false);
+                }
+                return true;
+            }, submitexthtlc.offerTx);
+            if (found)
+                throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("offer (%s) needs to have dfc htlc submitted first, but external htlc already submitted!",
+                        submitexthtlc.offerTx.GetHex()));
+            if (submitexthtlc.timeout < CICXSubmitEXTHTLC::MINIMUM_TIMEOUT)
+                throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Invalid parameters, argument \"timeout\" must be greater than %d", CICXSubmitEXTHTLC::MINIMUM_TIMEOUT - 1));
+        }         
+    }
+
+    CDataStream metadata(DfTxMarker, SER_NETWORK, PROTOCOL_VERSION);
+    metadata << static_cast<unsigned char>(CustomTxType::ICXSubmitEXTHTLC)
+             << submitexthtlc;
+
+    CScript scriptMeta;
+    scriptMeta << OP_RETURN << ToByteVector(metadata);
+
+    const auto txVersion = GetTransactionVersion(targetHeight);
+    CMutableTransaction rawTx(txVersion);
+
+    CTransactionRef optAuthTx;
+    std::set<CScript> auths;
```

Comment:
Do we need this an the subsequent auths.size() check? If not can be removed, auths in arg to GetAuthInputsSmart can be {} and auths.size() in conditional below can be removed.